src/topicButton.js

```diff
@@ -109,4 +133,65 @@ var TopicButton = GObject.registerClass({
         this._attentionSign.revealChild = value;
         this.notify('needs-attention');
     }
+
+    get revealed() {
+        return this._revealed;
+    }
+
+    set revealed(value) {
+        if ('_revealed' in this && this._revealed === value)
+            return;
+
+        this._revealed = value;
+        this.notify('revealed');
+    }
+
+    get sensitive() {
+        return this._sensitive;
+    }
+
+    set sensitive(value) {
+        if ('_sensitive' in this && this._sensitive === value)
+            return;
+
+        this._sensitive = value;
+        this.notify('sensitive');
+
+        if ('_sensitive' in this && !this._sensitive) {
+            this.get_style_context().remove_class('topic');
+            this.get_style_context().add_class('topic-insensitive');
+        } else if ('_sensitive' in this && this._sensitive) {
+            this.get_style_context().remove_class('topic-insensitive');
+            this.get_style_context().add_class('topic');
+        }
+    }
+
+    _dbusRegister() {
+        const objPath = Gio.Application.get_default().get_dbus_object_path();
+        const keys = Object.keys(Gio.Application.get_default()._windows);
+        const winName = keys[keys.length - 1].split('.');
+        const connId = winName[winName.length - 1];
+        this.objectPath = `${objPath}/window/${connId}/topic/${this._id}`;
```

Comment:
On hindsight, I think I gave you bad advice about this when I suggested to match it to the toolbox window object path. I think instead of `connId` here we should have the full app ID (although dots may need to be replaced by underscores to make it a valid object path.) Otherwise the quest has to guess or hardcode what the `connId` is for each app.

Later we should go back and change the object paths of the hack toolbox window objects, for this same reason, but that's outside the scope of this issue in any case.

Accessing `keys[keys.length - 1]` is a bit risky IMO anyway because a newer window could have been created in the meantime. I think you can get the app ID like this: `this.get_toplevel().target_app_id` (see `src/window.js`) but only after the topic button has been added to the widget hierarchy, so you might want to make `dbusRegister` public and call it from `addTopic`. (and indeed I'm not sure how I got there, but I ended up in a state where I had Sidetrack topics exported as `/com/endlessm/HackTopics/window/Fizzics/...`)